### PR TITLE
Fix webpack 4 issues with "start" script

### PR DESCRIPTION
**Pull request by wmarques**
Had some issues after updating a file with "yarn start" running, webpack crashed, I needed to bump some versions to make it work.

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
cool ! Hope it will fix https://github.com/jhipster/generator-jhipster/issues/7223

**Comment by pascalgrimaud** [comment]
@wmarques : just tested it, it doesn't resolve the issue :( `yarn start` is still broken for me

**Comment by wmarques** [comment]
@pascalgrimaud Strange, updated to the latest awesome-typescript-loader fixed this error for me..
Are you sure you ran `yarn install` ?

**Comment by pascalgrimaud** [comment]
Just tested again right now and nothing change for me -> it is still broken.
What I did:
- in generator-jhipster: `git checkout pr/7227` to switch on your branch
- generate a fresh project, so it launched `yarn install` and `yarn webpack:build` automatically
- check the `package.json` with `"awesome-typescript-loader": "4.0.0-0",`
- launch `./mvnw` and `yarn start` in 2 separate terminals
- create a file: `touch src/main/webapp/app/toto.html`
- :skull: crash! :skull:

**Comment by wmarques** [comment]
Well it seems to work on my computer... Really strange, yarn isn't playing you with his cache? Try to check in the `node_modules` that you have the correct dependencies installed, especially the typescript-loader and typescript.

**Comment by pascalgrimaud** [comment]
I checked everything and tested with an other computer at work, and it doesn't resolve the original issue.
If someone can:
- check the original issue https://github.com/jhipster/generator-jhipster/issues/7223
- check if this PR resolves it or not

I'm stuck with this issue with jhipster-registry. It's hard to launch `yarn start` each modifications. It seems like I'm back in 2008 without hot reload, I can't work without 'hot reload' that anymore !!!

**Comment by pascalgrimaud** [comment]
I checked everything and tested with an other computer at work, and it doesn't resolve the original issue.
If someone can:
- check the original issue https://github.com/jhipster/generator-jhipster/issues/7223
- check if this PR resolves it or not

I'm stuck with this issue with jhipster-registry. It's hard to launch `yarn start` each modifications. It seems like I'm back in 2008 without hot reload, I can't work without 'hot reload' anymore !!!

**Comment by gmarziou** [comment]
I confirm Pascal's report, it crashes also for me on windows 10 when adding a new js file.

<summary>I also noticed many errors before webpack start dev server though the app was working in browser</summary>
<details>
<pre>
ERROR in [at-loader] TS6053: File 'e:/lib.dom.d.ts' not found.                                                                                                                                
                                                                                                                                                                                              
ERROR in [at-loader] TS6053: File 'e:/lib.es2016.d.ts' not found.                                                                                                                             
                                                                                                                                                                                              
ERROR in [at-loader] TS2318: Cannot find global type 'Array'.                                                                                                                                 
                                                                                                                                                                                              
ERROR in [at-loader] TS2318: Cannot find global type 'Boolean'.                                                                                                                               
                                                                                                                                                                                              
ERROR in [at-loader] TS2318: Cannot find global type 'Function'.                                                                                                                              
                                                                                                                                                                                              
ERROR in [at-loader] TS2318: Cannot find global type 'IArguments'.                                                                                                                            
                                                                                                                                                                                              
ERROR in [at-loader] TS2318: Cannot find global type 'Number'.                                                                                                                                
                                                                                                                                                                                              
ERROR in [at-loader] TS2318: Cannot find global type 'Object'.                                                                                                                                
                                                                                                                                                                                              
ERROR in [at-loader] TS2318: Cannot find global type 'RegExp'.                                                                                                                                
                                                                                                                                                                                              
ERROR in [at-loader] TS2318: Cannot find global type 'String'.                                                                                                                                
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/app/admin/audits/audits.component.spec.ts:14:27                                                                                               
    TS2339: Property 'slice' does not exist on type 'string'.                                                                                                                                 
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/app/admin/audits/audits.component.spec.ts:18:15                                                                                               
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/app/admin/audits/audits.component.spec.ts:18:26                                                                                               
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/app/admin/audits/audits.component.spec.ts:25:24                                                                                               
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/app/admin/audits/audits.component.spec.ts:27:24                                                                                               
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/app/admin/audits/audits.component.spec.ts:118:29                                                                                              
    TS2339: Property 'length' does not exist on type '{}'.                                                                                                                                    
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/app/admin/audits/audits.component.spec.ts:131:29                                                                                              
    TS2339: Property 'length' does not exist on type '{}'.                                                                                                                                    
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:26:39                                                                                          
    TS2339: Property 'test' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:27:39                                                                                          
    TS2339: Property 'test' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:28:34                                                                                          
    TS2339: Property 'test' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:29:31                                                                                          
    TS2339: Property 'test' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:32:37                                                                                          
    TS2339: Property 'filter' does not exist on type '{}'.                                                                                                                                    
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:36:24                                                                                          
    TS2339: Property 'length' does not exist on type 'string'.                                                                                                                                
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:36:36                                                                                          
    TS2339: Property 'length' does not exist on type 'string'.                                                                                                                                
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:40:19                                                                                          
    TS2339: Property 'length' does not exist on type 'string'.                                                                                                                                
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:40:33                                                                                          
    TS2304: Cannot find name 'Math'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:43:39                                                                                          
    TS2304: Cannot find name 'Math'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:44:39                                                                                          
    TS2304: Cannot find name 'Math'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/account/password/password-strength-bar.component.ts:45:39                                                                                          
    TS2304: Cannot find name 'Math'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/user-management/user-management.component.ts:97:20                                                                                           
    TS2339: Property 'push' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/user-management/user-modal.service.ts:15:52                                                                                                  
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/user-management/user-modal.service.ts:16:20                                                                                                  
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/metrics/metrics.component.ts:33:13                                                                                                           
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/metrics/metrics.component.ts:39:13                                                                                                           
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/metrics/metrics.component.ts:48:47                                                                                                           
    TS2339: Property 'length' does not exist on type 'string'.                                                                                                                                
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/metrics/metrics.component.ts:72:13                                                                                                           
    TS2304: Cannot find name 'isNaN'.                                                                                                                                                         
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/health/health-modal.component.ts:29:28                                                                                                       
    TS2339: Property 'toFixed' does not exist on type 'number'.                                                                                                                               
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/health/health-modal.component.ts:31:42                                                                                                       
    TS2339: Property 'toFixed' does not exist on type 'number'.                                                                                                                               
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/health/health-modal.component.ts:36:20                                                                                                       
    TS2304: Cannot find name 'JSON'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/health/health-modal.component.ts:38:26                                                                                                       
    TS2339: Property 'toString' does not exist on type 'number'.                                                                                                                              
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/configuration/configuration.component.ts:24:17                                                                                               
    TS2304: Cannot find name 'Array'.                                                                                                                                                         
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/configuration/configuration.component.ts:25:42                                                                                               
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/configuration/configuration.component.ts:34:37                                                                                               
    TS2339: Property 'push' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/configuration/configuration.component.ts:34:42                                                                                               
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/configuration/configuration.service.ts:17:32                                                                                                 
    TS2339: Property 'push' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/configuration/configuration.service.ts:21:31                                                                                                 
    TS2339: Property 'sort' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/configuration/configuration.service.ts:27:37                                                                                                 
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/configuration/configuration.service.ts:34:26                                                                                                 
    TS2339: Property 'startsWith' does not exist on type 'string'.                                                                                                                            
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/configuration/configuration.service.ts:54:30                                                                                                 
    TS2339: Property 'push' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/audits/audits.component.ts:58:23                                                                                                             
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/audits/audits.component.ts:58:34                                                                                                             
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/audits/audits.component.ts:61:28                                                                                                             
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/audits/audits.component.ts:63:28                                                                                                             
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/audits/audits.component.ts:72:22                                                                                                             
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/audits/audits.component.ts:72:33                                                                                                             
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/audits/audits.component.ts:74:26                                                                                                             
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/admin/audits/audits.component.ts:96:20                                                                                                             
    TS2339: Property 'push' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/shared/alert/alert-error.component.ts:91:21                                                                                                        
    TS2339: Property 'push' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/blocks/interceptor/notification.interceptor.ts:12:9                                                                                                
    TS2555: Expected at least 2 arguments, but got 1.                                                                                                                                         
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/blocks/interceptor/notification.interceptor.ts:22:25                                                                                               
    TS2339: Property 'forEach' does not exist on type '{}'.                                                                                                                                   
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/layouts/profiles/profile.service.ts:10:26                                                                                                          
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/layouts/profiles/profile.service.ts:14:23                                                                                                          
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/layouts/profiles/profile.service.ts:23:59                                                                                                          
    TS2339: Property 'includes' does not exist on type '{}'.                                                                                                                                  
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/layouts/profiles/profile.service.ts:24:61                                                                                                          
    TS2339: Property 'includes' does not exist on type '{}'.                                                                                                                                  
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/user-route-access-service.ts:17:87                                                                                                       
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/user-route-access-service.ts:25:53                                                                                                       
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/user-route-access-service.ts:27:16                                                                                                       
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/user-route-access-service.ts:29:49                                                                                                       
    TS2339: Property 'length' does not exist on type '{}'.                                                                                                                                    
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/shared/login/login.component.ts:56:72                                                                                                              
    TS2339: Property 'test' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/shared/login/login.component.ts:56:110                                                                                                             
    TS2339: Property 'test' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/login/login.service.ts:18:20                                                                                                                  
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/helpers/mock-login.service.ts:23:53                                                                                                           
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/helpers/mock-language.service.ts:14:63                                                                                                        
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/helpers/mock-language.service.ts:34:55                                                                                                        
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/blocks/interceptor/errorhandler.interceptor.ts:14:92                                                                                               
    TS2339: Property 'indexOf' does not exist on type 'string'.                                                                                                                               
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/principal.service.ts:20:45                                                                                                               
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/principal.service.ts:21:16                                                                                                               
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/principal.service.ts:29:41                                                                                                               
    TS2339: Property 'length' does not exist on type '{}'.                                                                                                                                    
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/principal.service.ts:38:38                                                                                                               
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/principal.service.ts:40:20                                                                                                               
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/principal.service.ts:45:24                                                                                                               
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/principal.service.ts:48:24                                                                                                               
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/principal.service.ts:53:32                                                                                                               
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/principal.service.ts:61:20                                                                                                               
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/helpers/mock-principal.service.ts:14:59                                                                                                       
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/language/language.helper.ts:19:53                                                                                                             
    TS2304: Cannot find name 'document'.                                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/language/language.helper.ts:23:15                                                                                                             
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/language/language.helper.ts:24:16                                                                                                             
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/language/language.helper.ts:46:40                                                                                                             
    TS2304: Cannot find name 'document'.                                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/auth-jwt.service.ts:22:123                                                                                                               
    TS2339: Property 'bind' does not exist on type '(resp: any) => any'.                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/auth-jwt.service.ts:37:20                                                                                                                
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/auth-jwt.service.ts:39:20                                                                                                                
    TS2304: Cannot find name 'Promise'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/account.service.ts:11:36                                                                                                                 
    TS2304: Cannot find name 'Account'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/auth/account.service.ts:12:30                                                                                                                 
    TS2304: Cannot find name 'Account'.                                                                                                                                                       
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/blocks/interceptor/auth.interceptor.ts:11:50                                                                                                       
    TS2339: Property 'test' does not exist on type '{}'.                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/blocks/interceptor/auth.interceptor.ts:11:103                                                                                                      
    TS2339: Property 'startsWith' does not exist on type 'string'.                                                                                                                            
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/shared/util/request-util.ts:6:9                                                                                                                    
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/user/user.model.ts:11:19                                                                                                                      
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/user/user.model.ts:13:24                                                                                                                      
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/user/user.model.ts:28:30                                                                                                                      
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/main/webapp/app/core/user/user.model.ts:30:35                                                                                                                      
    TS2304: Cannot find name 'Date'.                                                                                                                                                          
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/helpers/spyobject.ts:7:21                                                                                                                     
    TS2304: Cannot find name 'Function'.                                                                                                                                                      
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/helpers/spyobject.ts:21:9                                                                                                                     
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/helpers/spyobject.ts:22:9                                                                                                                     
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/helpers/spyobject.ts:23:9                                                                                                                     
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
                                                                                                                                                                                              
ERROR in [at-loader] ./src/test/javascript/spec/helpers/spyobject.ts:31:13                                                                                                                    
    TS2304: Cannot find name 'Object'.                                                                                                                                                        
Child html-webpack-plugin for "index.html":                                                                                                                                                   
     1 asset                                                                                                                                                                                  
    Entrypoint undefined = index.html                                                                                                                                                         
    [./node_modules/html-webpack-plugin/lib/loader.js!./src/main/webapp/index.html] 875 bytes {0} [built]                                                                                     
           factory:394ms building:2991ms = 3385ms                                                                                                                                             
i ｢wdm｣: Failed to compile.                                                                                                                                                                   
[Browsersync] Proxying: http://localhost:9060                                                                                                                                                 
[Browsersync] Access URLs:                                                                                                                                                                    
 -------------------------------------                                                                                                                                                        
       Local: http://localhost:9000                                                                                                                                                           
    External: http://172.27.112.1:9000                                                                                                                                                        
 -------------------------------------                                                                                                                                                        
          UI: http://localhost:3001                                                                                                                                                           
 UI External: http://172.27.112.1:3001                                                                                                                                                        
 -------------------------------------                                                                                                                                                        
i ｢wdm｣: Compiling...                                                                                                                                                                         
                                                                                                                                                                                              
e:\projets\jhipster-projects\issues\wp4\node_modules\awesome-typescript-loader\src\instance.ts:73                                                                                             
        if (!compiler._tsInstances) {                                                                                                                                                         
               ^                                                                                                                                                                              
TypeError: Cannot read property '_tsInstances' of undefined                                                                                                                                   
    at resolveInstance (e:\projets\jhipster-projects\issues\wp4\node_modules\awesome-typescript-loader\src\instance.ts:73:16)                                                                 
    at e:\projets\jhipster-projects\issues\wp4\node_modules\awesome-typescript-loader\src\instance.ts:368:20                                                                                  
    at _err2 (eval at create (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack\node_modules\tapable\lib\HookCodeFactory.js:24:12), <anonymous>:26:1)                              
    at invalid (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack-dev-server\node_modules\webpack-dev-middleware\lib\context.js:76:7)                                              
    at context.compiler.plugin (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack-dev-server\node_modules\webpack-dev-middleware\lib\context.js:116:5)                             
    at _err0 (eval at create (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack\node_modules\tapable\lib\HookCodeFactory.js:24:12), <anonymous>:21:1)                              
    at e:\projets\jhipster-projects\issues\wp4\node_modules\browser-sync-webpack-plugin\index.js:26:5                                                                                         
    at AsyncSeriesHook.eval [as callAsync] (eval at create (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack\node_modules\tapable\lib\HookCodeFactory.js:24:12), <anonymous>:7:1) 
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack\node_modules\tapable\lib\Hook.js:35:21)                                  
    at Watching._go (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack\lib\Watching.js:40:32)                                                                                      
    at Watching._invalidate (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack\lib\Watching.js:164:9)                                                                              
    at watcher.compiler.watchFileSystem.watch (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack\lib\Watching.js:135:10)                                                           
    at watcher.wfs.watch (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack\lib\WatchIgnorePlugin.js:69:5)                                                                         
    at Watchpack.watcher.once (e:\projets\jhipster-projects\issues\wp4\node_modules\webpack\lib\node\NodeWatchFileSystem.js:43:4)                                                             
    at Object.onceWrapper (events.js:317:30)                                                                                                                                                  
    at emitTwo (events.js:126:13)                                                                                                                                                             
error An unexpected error occurred: "Command failed.                                                                                                                                          
Exit code: 1
</pre>                                                                                                                                                                         
</details>

**Comment by wmarques** [comment]
Ok then let's close this, I plan to use the ngtools/webpack instead of awesome-typescript-loader as discussed in #7231 so maybe it's not necessary to fix this..

**Comment by sendilkumarn** [comment]
@pascalgrimaud for time being, you can try this.

copy the webpack-common / webpack-dev /webpack-prod from the #7231 and use it straightaway. Just rename `tsconfig` to `tsconfig-aot` It should work seamlessly.

**Comment by deepu105** [comment]
@sendilkumarn lets switch to ts-loader its much faster than at-loader

**Comment by deepu105** [comment]
i'll push it on this branch

**Comment by deepu105** [comment]
Switched to ts-loader build time reduced from 16s to 9s on a default project, will later do some benchmarks with entities and stuff

**Comment by deepu105** [comment]
I think the speed can be still improved, i'll work on it further

**Comment by sendilkumarn** [comment]
Good work @deepu105 & Good Suggestion @sendilkumarn 😉

**Comment by sendilkumarn** [comment]
I think we should definitely let know that ngtools is not fast enough. Better to open an issue in ngtools repo. It is quite annoying

File: generators/client/templates/angular/webpack/webpack.dev.js.ejs [hunk 1]
```diff
@@ -79,12 +80,26 @@ module.exports = webpackMerge(commonConfig({ env: ENV }), {
...
+                loader: 'ts-loader',
+                    options: {
+                        transpileOnly: true,
+                        happyPackMode: true
+                    }
+                },
+                { loader: 'angular-router-loader' }
             ],
-            exclude: ['node_modules/generator-jhipster']
+            exclude: ['node_modules']
```
Comment (sendilkumarn) [inline generators/client/templates/angular/webpack/webpack.dev.js.ejs]: Is this giving the speed 🤔

Comment (deepu105) [reply generators/client/templates/angular/webpack/webpack.dev.js.ejs]: Nope, it doesn't seem needed, but i found a weird error in yarn start so need to check if thats from this

File: generators/client/templates/angular/webpack/webpack.dev.js.ejs [hunk 2]
```diff
@@ -79,12 +80,26 @@ module.exports = webpackMerge(commonConfig({ env: ENV }), {
...
+                { loader: 'cache-loader' },
+                {
+                    loader: 'thread-loader',
+                    options: {
+                        // there should be 1 cpu for the fork-ts-checker-webpack-plugin
+                        workers: require('os').cpus().length - 1
+                    }
+                },
+                {
+                loader: 'ts-loader',
```
Comment (sendilkumarn) [inline generators/client/templates/angular/webpack/webpack.dev.js.ejs]: identation?

File: generators/client/templates/angular/webpack/webpack.test.js.ejs [hunk 3]
```diff
@@ -33,13 +33,18 @@ module.exports = (WATCH) => ({
...
                 loader: 'tslint-loader',
                 exclude: /node_modules/
             },
             {
                 test: /\.ts$/,
-                loaders: ['awesome-typescript-loader', 'angular2-template-loader?keepUrl=true'],
+                use: [
+                    { loader: 'cache-loader' },
+                    { loader: 'ts-loader' },
+                    { loader: 'angular2-template-loader?keepUrl=true' }
```
Comment (sendilkumarn) [inline generators/client/templates/angular/webpack/webpack.test.js.ejs]: remove the query with options

**Pull request merged**